OloEngine/src/OloEngine/Serialization/MeshBinarySerializer.cpp

```
@@ -0,0 +1,1254 @@
+            if (compressedSize > MAX_COMPRESSED_SIZE)
+            {
+                OLO_CORE_ERROR("MeshBinarySerializer::Read: compressed payload size {} exceeds limit in '{}'",
+                               compressedSize, path.string());
+                return nullptr;
+            }
+
+            std::vector<u8> compressedData(compressedSize);
+            ReadBytes(in, compressedData.data(), compressedSize);
+
+            // Validate CRC32 checksum
+            if (header.Checksum != 0)
```

Clang (14.0.6)

[note] +2, including nesting penalty of 1, nesting level increased to 2

(clang)